### DISCUSSION TITLE
Editing to config.php does not persist after reboot of server

### DISCUSSION THREAD

**Discussion by LarsEggimann** [nextcloud/all-in-one#3585] [Questions] [answered]
I have the nexcloud-aio running on a raspberry pi4 and would like to use preview generator with imaginary and therefore need to edit the config.php file. I tried the command suggested in [this FAQ](https://github.com/nextcloud/all-in-one#how-to-edit-nextclouds-configphp-file-with-a-texteditor) but after `sudo reboot now` the config.php file is reset.

what am I doing wrong?

**Comment by szaimen** [comment]
Hi, what values exactly did you try ro adjust manually? Also to what were they changed after a restart?

**Reply by LarsEggimann** [reply]
Hi, thanks for the answer.

I tried to add those values:
```
'enabledPreviewProviders' => 
  array (
    0 => 'OC\\Preview\\Imaginary',
    ... 
   ),
  'preview_imaginary_url' => 'http://127.0.0.1:9191',
```

I was following [this tutorial](https://okxo.de/speed-up-nextcloud-preview-generation-with-imaginary/) and changed the port to 9191 as on 9000 I have portainer running. I ran `sudo docker exec -it nextcloud-aio-nextcloud sudo -u www-data php occ preview:generate-all -vvv ` and noticed that in the log of the imaginary-docker (`sudo docker logs -f nextcloud-aio-imaginary`) there were no entries. I checked the config.php again and my added entries were missing again ... after several retries (im sure i saved the file and also looked at it again after changeing) as soon as i reboot the raspberry to apply the changes the config.php is reset. 

Thank you for your help, really appriciate it. :)

**Reply by Zoey2936** [reply, marked as answer]
can I ask why you not simply open the AIO interface and enable imaginary?

**Reply by LarsEggimann** [reply]
Hi Zoey

I did not know one could do that. Where would I enable imaginary in the AIO interface (don't see it):
![grafik](https://github.com/nextcloud/all-in-one/assets/82716150/8a39ae1f-2867-4f6d-99e5-26606d1c001d)


Thanks in advance!

**Reply by szaimen** [reply]
@thelittleicebear did you already try to scroll down on this page?

**Reply by LarsEggimann** [reply]
ah i see .... i'm so sorry ...

I will try it and let you know it it works now. thanks again ... ^^

**Reply by LarsEggimann** [reply]
thank you so much, it is working now. I can see the logs and the previewgenerator is working now. I was not aware that I can add imaginary in the nexcloud AIO panel.

Generally I find it a bit confusing when searching for tutorials or guides, most of the time it is not done with the AIO methode and i could not find any description/guide of the Imaginary container on the github page.

thanks again, and sorry for the dumb question

**Reply by LarsEggimann** [reply]
after some running time I now get the following error:
```
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 4096 bytes) in /var/www/html/lib/private/legacy/OC_Image.php on line
```

In my docker-compose.yml i have set the enviroment variable `NEXTCLOUD_MEMORY_LIMIT=512M`, would this error go away if I increase this memory limit? As i only have 2GB of ram on my raspbery im not sure if i should increase this?

Thanks in advance.
Cheers,
Lars

**Reply by szaimen** [reply]
> As i only have 2GB of ram on my raspbery

I guess this is your main problem then...

> In my docker-compose.yml i have set the enviroment variable `NEXTCLOUD_MEMORY_LIMIT=512M`, would this error go away if I increase this memory limit?

You could try that but this will likely not help

**Reply by LarsEggimann** [reply]
I see. so the preview generation cannot finish if I dont allocate more RAM? Or where does that error come from?

Could the low RAM also be a reason for the pi to become unresponsive? i have synced a few big pictures and when i try to browse that folder in the browser app the whole system gets unresponsive (no ssh connection or portainer possible) upon which i have to pull the plug to get it working again.
I know this is very unspecific but im glad if you could share your opinion.

**Reply by TherionIM** [reply]
Hello,

I'm having a similar issue where I need to manually edit the config.php instead of enabling the imaginary container as I would like to use a "remote" instance of imaginary running on another machine on my local network. I also followed the documentation and make the changes and used the `preview_imaginary_url` to point to the instance. Following the edit I choose to stop and restart the containers but the setting changes disappear. Any ideas?

Thanks

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-keep-disabled-apps
